sdk/storage/src/table/continuation_token.rs

```diff
@@ -0,0 +1,33 @@
+use azure_core::errors::AzureError;
+use http::HeaderMap;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ContinuationToken {
+    pub(crate) query_path: String,
+    pub(crate) partition_key: String,
+    pub(crate) row_key: String,
+}
+
+impl ContinuationToken {
+    pub(crate) fn parse_from_headers_optional(
+        query_path: &str,
+        headers: &HeaderMap,
+    ) -> Result<Option<Self>, AzureError> {
+        const HEADER_NEXTPARTITIONKEY: &str = "x-ms-continuation-NextPartitionKey";
+        const HEADER_NEXTROWKEY: &str = "x-ms-continuation-NextRowKey";
+
+        Ok(
+            if headers.contains_key(HEADER_NEXTPARTITIONKEY)
```

Comment:
nit: can you move this to outside the `Ok()` and put it in a variable? This is somewhat hard to read.